tests/rptest/tests/partition_movement_test.py

```diff
@@ -500,3 +553,22 @@ def partition_ready():
         assert admin.get_partitions(name, 0)['status'] == "in_progress"
         rpk.delete_topic(name)
         assert name not in rpk.list_topics()
+
+    @cluster(num_nodes=5)
+    def test_shadow_indexing(self):
+        """
+        Test interaction between the shadow indexing and the partition movement.
+        Partition movement generate partitions with different revision-ids and the
+        archival/shadow-indexing subsystem is using revision to generate unique object
+        keys inside the remote storage.
+        """
+        self.start_redpanda(num_nodes=3)
+        spec = TopicSpec(name="topic", partition_count=1, replication_factor=3)
+        self.client().create_topic(spec)
+        self.topic = spec.name
+        self.start_producer(1)
+        self.start_consumer(1)
+        self.await_startup()
+        for _ in range(25):
+            self._move_and_verify()
+        self.run_validation(enable_idempotence=False, consumer_timeout_sec=45)
```

Comment:
Isn't it possible that on partition movement hides errors by the previous one? I wonder if we should run validation after every partition movement

Reply:
good point @LenaAn. that was the intention of `move_and_verify` but it looks like this particular test is doing _extra_ verification that should be repeated for each move.

Reply:
I tried this but it takes too much time to finish without triggering any new errors. I guess I'll keep it that way by now.